tests/Opus/Search/Common/IndexerTest.php

```diff
@@ -0,0 +1,961 @@
+		$this->indexer->removeDocumentsFromIndex( array( $document ) );
+
+		$this->assertEquals( 0, $this->_getNumberOfIndexDocs() );
+	}
+
+	public function testRemoveDocumentByIdFromIndex() {
+		$this->_addOneDocumentToIndex();
+
+		$this->assertEquals( 1, $this->_getNumberOfIndexDocs() );
+
+		$this->indexer->removeDocumentsFromIndexById( $this->document_id );
+
+		$this->assertEquals( 0, $this->_getNumberOfIndexDocs() );
+	}
+
+	public function testRemoveDocumentArrayByIdFromIndex() {
+		$this->_addOneDocumentToIndex();
+
+		$this->assertEquals( 1, $this->_getNumberOfIndexDocs() );
+
+		$this->indexer->removeDocumentsFromIndexById( array( $this->document_id ) );
+
+		$this->assertEquals( 0, $this->_getNumberOfIndexDocs() );
+	}
+
+	/**
+	 * @expectedException \InvalidArgumentException
+	 */
+	public function testRemoveNullFromIndex() {
+		$this->_addOneDocumentToIndex();
+
+		$this->indexer->removeDocumentsFromIndex( null );
+	}
+
+	/**
+	 * @expectedException \InvalidArgumentException
+	 */
+	public function testRemoveMissingDocumentFromIndex() {
+		$this->_addOneDocumentToIndex();
+
+		$this->indexer->removeDocumentsFromIndex();
+	}
+
+	public function testDeleteAllDocsFromNonEmptyIndex() {
+		$this->_addOneDocumentToIndex();
+		$this->assertEquals( 1, $this->_getNumberOfIndexDocs() );
+
+		$this->indexer->removeAllDocumentsFromIndex();
+
+		$this->assertEquals( 0, $this->_getNumberOfIndexDocs() );
+	}
+
+	public function testDeleteAllDocsFromEmptyIndex() {
+		$this->indexer->removeAllDocumentsFromIndex();
+
+		$this->assertEquals( 0, $this->_getNumberOfIndexDocs() );
+	}
+
+	public function testDeleteDocsByMatchingQuery() {
+		$this->markTestSkipped( 'not supported by Opus_Search API' );
+
+		/*		$this->_addOneDocumentToIndex();
+
+				$this->assertEquals( 1, $this->_getNumberOfIndexDocs() );
+
+				$queryString = 'id:' . $this->document_id;
+				$this->indexer->deleteDocsByQuery( $queryString );
+				$this->indexer->commit();
+
+				$this->assertEquals( 0, $this->_getNumberOfIndexDocs() );*/
+	}
+
+	public function testDeleteDocsByNonMatchingQuery() {
+		$this->markTestSkipped( 'not supported by Opus_Search API' );
+
+		/*		$this->_addOneDocumentToIndex();
+
+				$this->assertEquals( 1, $this->_getNumberOfIndexDocs() );
+
+				$nonExistentDocID = $this->document_id + 1;
+				$queryString      = 'id:' . $nonExistentDocID;
+				$this->indexer->deleteDocsByQuery( $queryString );
+				$this->indexer->commit();
+
+				$this->assertEquals( 1, $this->_getNumberOfIndexDocs() );*/
+	}
+
+	public function testDeleteDocsByInvalidQuery() {
+		$this->markTestSkipped( 'not supported by Opus_Search API' );
+
+		/*		$this->setExpectedException( 'Opus_SolrSearch_Index_Exception' );
+				$this->indexer->deleteDocsByQuery( 'id:' );*/
+	}
+
+	public function testCommit() {
+		$this->markTestSkipped( 'not supported by Opus_Search API' );
+
+		/*		$this->indexer->commit();*/
+	}
+
+	public function testOptimize() {
```

Comment:
We have scripts that use optimize and there are plans for having an optimize button in the administration web interface. Is it possible to run a optimization with Solarium through the new API?